apps/src/weblab2/helpers/aiTutorStructuredResponseHelper.ts

```diff
@@ -137,3 +141,110 @@ export const formatExplanationResponse = (response: any): string => {
   }
   return formattedResponse;
 };
+
+type AiTutorCodeFile = {
+  name: string;
+  contents: string;
+};
+
+type AcceptRejectFormattedResponse = {
+  explanation: string;
+  code: AiTutorCodeFile[];
+};
+
+// Parsed json comes in as 'any', but it follows the structure defined in acceptRejectJsonSchema.
+export const formatAcceptRejectResponse = (
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  response: any
+): AcceptRejectFormattedResponse => {
+  return {
+    explanation: response.explanation || '',
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    code: response.code.map((codeFile: any) => ({
+      name: codeFile.filename,
+      contents: codeFile.sourceCode,
+    })),
+  };
+};
+
+export const getMergedAiTutorCodeWithSource = (
+  code: AiTutorCodeFile[],
+  source: MultiFileSource,
+  aiTutorVersionFiles: ProjectFile[]
+): MultiFileSource => {
+  // Helper function to get folder depth (distance from root)
+  const getFolderDepth = (folderId: string): number => {
+    if (folderId === '0') return 0; // Root folder
+    const folder = source.folders[folderId];
+    if (!folder) return 0;
+    return 1 + getFolderDepth(folder.parentId);
+  };
+
+  // Create copy of the source. Set openFiles to empty array to close all files for now.
+  const updatedSource: MultiFileSource = {
+    ...source,
+    files: {},
+    folders: {...source.folders},
+    openFiles: [],
+  };
+  // Set all files as inactive by creating new file objects.
+  Object.keys(source.files).forEach(fileId => {
+    updatedSource.files[fileId] = {
+      ...source.files[fileId],
+      active: false,
+    };
+  });
+
+  // For each AI code file, find and replace the matching file if there is a matching file.
+  code.forEach((aiFile: AiTutorCodeFile) => {
+    // First check active file is the same as the AI code file.
+    const activeFile = getActiveFileForSource(source);
+    if (activeFile?.name === aiFile.name) {
+      // Active file is the same as the AI code file - replace it.
```

Comment:
If there are multiple files with the same name, then if the file name matches and it's currently active, changes are high that this is the one the user wants to update with the help of AI.